packages/native/src/useLinking.tsx

```diff
@@ -592,7 +596,8 @@ export default function useLinking(
               // We couldn't find an existing entry to go back to, so we'll go back by the delta
               // This won't be correct if multiple routes were pushed in one go before
               // Usually this shouldn't happen and this is a fallback for that
-              await history.go(historyDelta);
+              // Add Math.max to guard going to < 0 index
+              await history.go(Math.max(historyDelta, 0));
```

Comment:
guarding the historyDelta here. Sometimes it returns -1, which makes it routes to outside the app when using initialRouteName (see issue #10398 for screenshot).

My thought process:
When using initialRouteName, we don't store the initialRouteName in the browser's history
This means when historyDelta = -1, it will try to go to the previous page in browser's history (instead of our specified route name)
To fix this, I added a guard so we make sure we never navigate outside App

Reply:
This branch of code (`else if (historyDelta < 0) {`) only executes when the `historyDelta` is negative, it doesn't make sense to guard it here - it's ALWAYS going to call `history.go(0)` when it comes here. It's the same as not having this branch of the condition at all.

If this shouldn't happen, then this code should be removed and tested to see if this doesn't break anything.